src/WebDavClient.php

```diff
@@ -150,4 +151,53 @@ public function setCustomSetting(array $connectionConfig, string $accessToken):
             $this->addCurlSetting($setting, $value);
         }
     }
+
+    /**
+     * @param string $pattern
+     * @param string|null $limit
+     * @param string|null $url
+     * @return array<string, array<int, array<string, string|null|object>>>
+     * @throws HttpException
+     * @throws SabreClientException
+     * @throws SabreClientHttpException
+     * @throws \DOMException
+     */
+    public function sendReportRequest(string $pattern, ?string $limit, ?string $url = ''): array
```

Comment:
Even if we need `$limit` as a string in the XML it feels weird to make this parameter in the function an string, I think it should be `int` and if the XML strictly needs a string, then cast it to string later.
Using `string` here will make it much more error-prone. Same applies to higher level functions